frontends/concrete-python/concrete/fhe/mlir/processors/assign_bit_widths.py

```diff
@@ -111,14 +111,17 @@ def apply(self, graph: Graph):
         for node, bit_width in bit_widths.items():
             assert isinstance(node.output.dtype, Integer)
             new_bit_width = model[bit_width].as_long()
-
-            if node.output.is_clear:
-                new_bit_width += 1
-
-            node.properties["original_bit_width"] = node.properties.get(
+            original_bit_width = node.properties.get(
                 "bit_width_hint",
                 node.output.dtype.bit_width,
             )
+
+            if node.output.is_clear:
+                new_bit_width = original_bit_width
+                if not node.output.dtype.is_signed:
+                    new_bit_width += 1
```

Comment:
Why +1 here?

Reply:
We're doing a signed extension to clear values and it can cause problems.

`3` in 2 bits is `11` but because we treat it as signed by default, it'll be expanded to `111...111` which isn't `3`. By adding one more bit to unsigned values, we can avoid this.